src/metabase/upload.clj

```diff
@@ -678,7 +676,53 @@
 ;;; |  hydrate based_on_upload for FE
 ;;; +--------------------------------
 
-(mi/define-simple-hydration-method based-on-upload
+(defn uploadable-table-ids
+  "Returns the subset of table ids where the user can upload to the table."
+  [table-ids]
+  (if (empty? table-ids)
+    #{}
+    (let [tables (t2/hydrate (t2/select :model/Table :id [:in table-ids]) :db)]
+      (set (keep (fn [t]
+                   (when (can-upload-to-table? (:db t) t)
+                     (:id t)))
+                 tables)))))
+
+(defn- no-joins?
```

Comment:
```suggestion
(defn- query-no-joins?
```

IMO, if the object that a helper function operates on is not the main object of the namespace, we should prefix it.

For example, it doesn't give much context if you call `(upload/no-joins ...)` since `upload` doesn't have a join property by itself.

Reply:
I can't see much benefit of applying that rule for private helper functions like this one which are only used in one function. I would agree with your example though, no one should use uploads/no-joins?without renaming it appropriately or moving it to an appropriate namespace.

Reply:
Agree on not needing a prefix for a private method (implicit in the arg name), but I do find this a strange name for a predicate. Having a `has-joins?` method that just gets negated at the call site feels a bit clearer to me.